avm/src/vm/opcode.rs

```diff
@@ -2682,12 +2677,12 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
         hm.register_out(args[0], i, CLOSURE_ARG_MEM_START + 1);
         filters.push(subhandler.clone().run(hm));
       }
-      let mut hms = join_all(filters).await;
+      let hms = join_all(filters).await;
       let mut idxs = vec![];
-      for i in 0..len {
-        let hm = &mut hms[i];
+      for (i, hm) in hms.into_iter().enumerate() {
+        // let hm = &mut hms[i];
         let val = hm.read_fixed(CLOSURE_ARG_MEM_START);
-        hm.drop_parent();
+        hm.drop_parent(); // this drops `hm`
```

Comment:
you mean `hm` parrent which seems self explanatory?

Reply:
with the change I made to `drop_parent`'s `self` parameter, it also drops `hm`. I can revert `drop_parent` back though

Reply:
We decided not to revert `drop_parent` so that it matches the signature of the new `drop_parent_async`